internal/node/component/decentralized/relation.go

```diff
@@ -0,0 +1,292 @@
+	IsRequired  bool        `json:"isRequired"`
+	Type        string      `json:"type"`
+	Value       interface{} `json:"value"`
+	Description string      `json:"description"`
+}
+
+type Endpoint struct {
+	URL           *ConfigDetail `json:"network"`
+	HTTPHeaders   *ConfigDetail `json:"http_headers"`
+	HTTP2Disabled *ConfigDetail `json:"http2_disabled"`
+}
+
+type Parameters struct {
+	BlockHeightStart      *ConfigDetail `json:"block_height_start"`
+	BlockHeightTarget     *ConfigDetail `json:"block_height_target"`
+	BlockNumberStart      *ConfigDetail `json:"block_number_start"`
+	BlockNumberTarget     *ConfigDetail `json:"block_number_target"`
+	RPCThreadBlocks       *ConfigDetail `json:"rpc_thread_blocks"`
+	RPCBatchBlocks        *ConfigDetail `json:"rpc_batch_blocks"`
+	RPCBatchReceipts      *ConfigDetail `json:"rpc_batch_receipts"`
+	RPCBatchBlockReceipts *ConfigDetail `json:"rpc_batch_block_receipts"`
+	APIKey                *ConfigDetail `json:"api_key"`
+}
+
+type workerConfig struct {
+	WorkerID     *ConfigDetail `json:"worker_id"`
+	Network      ConfigDetail  `json:"network"`
+	Worker       ConfigDetail  `json:"worker"`
+	EndpointID   *ConfigDetail `json:"endpoint_id"`
+	Endpoint     *Endpoint     `json:"endpoint"`
+	IPFSGateways *ConfigDetail `json:"ipfs_gateways"`
+	Parameters   *Parameters   `json:"parameters"`
+}
+
+// NetworkToWorkersMap is a map of network to workers.
+var NetworkToWorkersMap = map[network.Network][]worker.Worker{
+	network.Ethereum: {
+		worker.Aave,
+		worker.Core,
+		worker.Curve,
+		worker.ENS,
+		worker.Highlight,
+		worker.Lido,
+		worker.Looksrare,
+		worker.Oneinch,
+		worker.OpenSea,
+		worker.Optimism,
+		worker.RSS3,
+		worker.Stargate,
+		worker.Uniswap,
+	},
+	network.Arweave: {
+		worker.Mirror,
+		worker.Momoka,
+		worker.Paragraph,
+	},
+	network.Farcaster: {
+		worker.Core,
+	},
+	network.Polygon: {
+		worker.Aave,
+		worker.Aavegotchi,
+		worker.Core,
+		worker.Curve,
+		worker.Highlight,
+		worker.IQWiki,
+		worker.Lens,
+		worker.Matters,
+		worker.Stargate,
+	},
+	network.Crossbell: {
+		worker.Crossbell,
+	},
+	network.Avalanche: {
+		worker.Aave,
+		worker.Curve,
+		worker.Stargate,
+	},
+	network.Base: {
+		worker.Aave,
+		worker.Core,
+		worker.Stargate,
+	},
+	network.Optimism: {
+		worker.Aave,
+		worker.Core,
+		worker.Curve,
+		worker.Highlight,
+		worker.KiwiStand,
+		worker.Stargate,
+	},
+	network.Arbitrum: {
+		worker.Aave,
+		worker.Core,
+		worker.Curve,
+		worker.Highlight,
+		worker.Stargate,
+	},
+	network.VSL: {
+		worker.Core,
+		worker.RSS3,
+		worker.VSL,
+	},
+	network.SatoshiVM: {
+		worker.Core,
+		worker.SAVM,
+		worker.Uniswap,
+	},
+	network.BinanceSmartChain: {
+		worker.Core,
+		worker.Stargate,
+	},
+	network.Gnosis: {
+		worker.Core,
+		worker.Curve,
+	},
+	network.Linea: {
+		worker.Core,
+		worker.Stargate,
+		worker.Uniswap,
+	},
+}
+
+// NetworkArchToConfigMap is a map of network arch to config.
+var NetworkArchToConfigMap = map[network.Source]workerConfig{
```

Comment:
Would it be better to list each worker’s configuration information separately for clarity and maintainability? @HenryQW

Reply:
we have 23 workers now and the amount is growing, may have too many duplicated lines 🧐

Reply:
for instance, all ethereum sources return the IPFSGateways configuration, but some workers actually do not need it. This could confuse users or lead to unnecessary operational costs.

Reply:
yes, adding too many special cases would be hard to maintain.

@HenryQW